dig_test.go

```diff
@@ -3049,3 +3049,107 @@ func TestCanVisualizeError(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkProvideCycleDetection(b *testing.B) {
```

Comment:
Yes, thank you. We should have this anyway. Perhaps we should land the benchmark first as a separate PR so we have them on master/dev and can more easily check against various optimization implementations?

Reply:
Resolving this because I added this Benchmark in #210, although our difference in results is yet unexplained.